### PR TITLE
Angular 11

### LINKED ISSUE #945: Support angular 11
**Is your feature request related to a problem? Please describe.**

Angular (& CLI) 11 has been released, and should be supported by the IDS angular component library.

**Describe the solution you'd like**

Support for angular 11

### PR DESCRIPTION
**Pull request by bthharper**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

This PR updates the component library to angular and angular/cli version 11.  In additional several other libraries have also been updated, specifically karma-coverage and eslint. 

Recommended to build, test and run the example application.

**Related github/jira issue (required)**:
Fixes #945

**Steps necessary to review your pull request (required)**:

Clone the angular-11 branch:

```powershell
npm i
npm run build
npm run lint
npm run test
npm run start
```

Verify the demo app on http://localhost:4200/ids-enterprise-ng-demo/.

For versioning we should make this a 9.0 along with the "strict compiler" changes.

### CODE REVIEW

File: projects/ids-enterprise-ng/src/lib/accordion/soho-accordion-header.component.ts [hunk 1]
```diff
@@ -24,8 +24,12 @@ export class SohoAccordionHeaderComponent implements AfterViewInit {
   jQueryElement: JQuery;
 
   @HostBinding('class.is-disabled') isDisabled = false;
-  @HostBinding('style.display') get isBlockDisplay() { return 'block'; }
-  @HostBinding('class.accordion-header') get isAccordionHeader() { return true; }
+  @HostBinding('style.display') get isBlockDisplay() {
+ return 'block';
+}
+  @HostBinding('class.accordion-header') get isAccordionHeader() {
```
Comment (tmcconechy) [inline projects/ids-enterprise-ng/src/lib/accordion/soho-accordion-header.component.ts:30]: Should this be:
```
  @HostBinding('class.accordion-header') get isAccordionHeader() {
    return true;
  }
```

If thats a rule im not sure why it doesnt line up.

**Review by tmcconechy — CHANGES_REQUESTED** [review]
This looks ok but its going to badly conflict with all the stuff on https://github.com/infor-design/enterprise-ng/pull/939

I think we should test that one and merge it and then update this branch again. I made som e other changes to get the editor to really respect the same errors in the tsconfigs and i see you also cleaned that up.

Plus other changes may conflict?

File: projects/ids-enterprise-ng/src/lib/accordion/soho-accordion-header.component.ts (hunk 1 above)
Comment (bthharper) [reply projects/ids-enterprise-ng/src/lib/accordion/soho-accordion-header.component.ts:30]: Yes - it only complains about the lack of newlines, I thought I fixed the formatting when I added the newline in but after doing 200 of them I must have missed one ;-)

**Comment by bthharper** [comment]
@tmcconechy  - I fixed all the tsconfig file to match those produced by the schematics for an angular project, they are remove the `tsconfig.base.json` and put back the `tsconfig.json`.

The `tslint` to `eslint` are quite invasive and possibly will cause conflicts.  Maybe we can try a test merge between the branches?

**Comment by tmcconechy** [comment]
Screen Shot 2020-11-23 at 12 42 52 PM

Looks like 80 conflicts. I think we need to fix the config.json ones as each file i looked at has a few hundred errors because it cant read the tsconfig files correctly.

I think either two choices

1) merge https://github.com/infor-design/enterprise-ng/pull/939 then apply it on top this and fix the problems. If you fix the tsconfig merges
2) skip the eslint for now and  merge https://github.com/infor-design/enterprise-ng/pull/939 then apply it on top this and fix the problems

the one thing i dont want to do is redo https://github.com/infor-design/enterprise-ng/pull/939 again as it was several weeks of work

80 merges seems managable.

**Comment by bthharper** [comment]
@tmcconechy  - add your changes to trunk, I can look at the conflicts with the ng-11 work.  I would imagine most of the tsconfig changes will be to keep the ones in this PR, just need to set the strict flag on again.

**Comment by tmcconechy** [comment]
Sounds good. Does that mean that the PR looks ok @bthharper ?

File: projects/ids-enterprise-ng/src/lib/accordion/soho-accordion-header.component.ts (hunk 1 above)
Comment (tmcconechy) [reply projects/ids-enterprise-ng/src/lib/accordion/soho-accordion-header.component.ts:30]: I just mean it looks like its like this....
```
  @HostBinding('style.display') get isBlockDisplay() {
 return 'block';
}
```

but maybe should be 
```
  @HostBinding('style.display') get isBlockDisplay() {
    return 'block';
   }
```

But no need to go back and find all of these... Maybe just fix a couple if you see them

**Comment by tmcconechy** [comment]
@bthharper ok i merged the big one in. Conflicts are "ready"

**Comment by tmcconechy** [comment]
@bthharper - Think we can wrap these fixes up by Tuesday next week? Want to release a 9.x then early next week and some interest in making NG 11 available. Or let me know if you want help if you cant do it?

**Comment by bthharper** [comment]
@tmcconechy - might be a push as I need to start the merge again,  Will probably revert and apply again, let me see how far I get.

File: projects/ids-enterprise-ng/src/lib/accordion/soho-accordion-header.component.ts (hunk 1 above)
Comment (bthharper) [reply projects/ids-enterprise-ng/src/lib/accordion/soho-accordion-header.component.ts:30]: Fixed

**Comment by bthharper** [comment]
@tmcconechy - can you review and if you're happy it should be ready to go.  Needs the version updating to 9 I think though.

**Review by tmcconechy — APPROVED** [review]
Looks Great. Thanks for doing this @bthharper

**Review by pwpatton — APPROVED** [review]

**Review by EdwardCoyle — APPROVED** [review]

**Pull request merged**